### ISSUE TITLE
Floods of `Cannot read property 'close' of null` on timeouts occurring occasionally

### ISSUE THREAD

**Issue by prawnsalad** [sockjs/sockjs-node#197]
~~~
TypeError: Cannot read property 'close' of null
    at WebSocketReceiver.heartbeat_timeout (/root/transport/node_modules/sockjs/lib/trans-websocket.js:157:26)
    at null._onTimeout (/root/transport/node_modules/sockjs/lib/trans-websocket.js:75:24)
    at Timer.listOnTimeout (timers.js:92:15)
~~~

**Comment by tmds** [comment]
Probably related to https://github.com/sockjs/sockjs-node/issues/185

**Comment by tmds** [comment]
What version of sockjs-node are you using?

**Comment by prawnsalad** [comment]
Using 0.3.16. Looks like it might be the same issue as you mentioned

**Comment by domderen** [comment]
I have the same problem with `sockjs@0.3.16` used inside `webpack-dev-server`. Stack trace:

```
/lightning-ui/node_modules/webpack-dev-server/node_modules/sockjs/lib/trans-websocket.js:157
      return this.session.close(3000, 'No response from heartbeat');
                         ^

TypeError: Cannot read property 'close' of null
    at WebSocketReceiver.heartbeat_timeout (/lightning-ui/node_modules/webpack-dev-server/node_modules/sockjs/lib/trans-websocket.js:157:26)
    at null._onTimeout (/lightning-ui/node_modules/webpack-dev-server/node_modules/sockjs/lib/trans-websocket.js:75:24)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by prewk** [comment]
This is becoming a real issue for us, on large projects the dev server can be pretty slow to restart. Do you have any ideas?

**Comment by brycekahle** [comment]
Fixed by #179

**Comment by brycekahle** [comment]
Released in 0.3.17
